packages/init/src/SliceMachineInitProcess.ts

```diff
@@ -135,12 +135,20 @@ export class SliceMachineInitProcess {
 				"Repository selection must be available through context to proceed",
 			);
 
+			await this.finishCoreDependenciesInstallation();
+			await this.upsertSliceMachineConfigurationAndStartPluginRunner();
+
+			const isLoginRequired = await this.checkIsLoginRequired();
+			if (isLoginRequired) {
+				await this.loginAndFetchUserData();
+				if (this.context.repository.exists) {
+					this.validateWriteAccess();
+				}
+			}
 			if (!this.context.repository.exists) {
 				await this.createNewRepository();
```

Comment:
How long do you think this take?

Reply:
the `validateWriteAccess` method?

Reply:
the await this.createNewRepository() method?